distiller.nf

```diff
@@ -451,39 +451,60 @@ process map_parse_sort_chunks {
     script:
     // additional mapping options or empty-line
     def mapping_options = params['map'].get('mapping_options','')
+    def trim = params['map'].get('trim','')
 
     def dropsam_flag = params['parse'].get('make_pairsam','false').toBoolean() ? '' : '--drop-sam'
     def dropreadid_flag = params['parse'].get('drop_readid','false').toBoolean() ? '--drop-readid' : ''
     def dropseq_flag = params['parse'].get('drop_seq','false').toBoolean() ? '--drop-seq' : ''
-    def keep_unparsed_bams_command = ( 
-        params['parse'].get('keep_unparsed_bams','false').toBoolean() ? 
+    def keep_unparsed_bams_command = (
+        params['parse'].get('keep_unparsed_bams','false').toBoolean() ?
         "| tee >(samtools view -bS > ${library}.${run}.${ASSEMBLY_NAME}.${chunk}.bam)" : "" )
     def parsing_options = params['parse'].get('parsing_options','')
 
-    //def bwa_threads = Math.max(1, 
+    //def bwa_threads = Math.max(1,
     //                      ((((task.cpus as int)*0.6).round()) as int))
     //def sorting_threads = Math.max(1, (task.cpus as int)-bwa_threads)
-    // Since bwa and sort operate on the same pipe, they do not use their 
+    // Since bwa and sort operate on the same pipe, they do not use their
     // cores simultaneously and it is safe to give both of them all cores.
     def bwa_threads = (task.cpus as int)
     def sorting_threads = (task.cpus as int)
 
-    """
-    TASK_TMP_DIR=\$(mktemp -d -p ${task.distillerTmpDir} distiller.tmp.XXXXXXXXXX)
-    touch ${library}.${run}.${ASSEMBLY_NAME}.${chunk}.bam 
-    bwa mem -t ${bwa_threads} ${mapping_options} -SP ${bwa_index_base} ${fastq1} ${fastq2} \
-        ${keep_unparsed_bams_command} \
-        | pairtools parse ${dropsam_flag} ${dropreadid_flag} ${dropseq_flag} \
-            ${parsing_options} \
-            -c ${chrom_sizes} \
-            | pairtools sort --nproc ${sorting_threads} \
-                             -o ${library}.${run}.${ASSEMBLY_NAME}.${chunk}.pairsam.${suffix} \
-                             --tmpdir \$TASK_TMP_DIR \
-            | cat
+    if(trim)
+        """
+        TASK_TMP_DIR=\$(mktemp -d -p ${task.distillerTmpDir} distiller.tmp.XXXXXXXXXX)
+        touch ${library}.${run}.${ASSEMBLY_NAME}.${chunk}.bam
+
+        fastp --adapter_sequence=AGATCGGAAGAGCACACGTCTGAACTCCAGTCA --adapter_sequence_r2=AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT \
```

Comment:
The second suggestion would be to try avoiding this code duplication. The solution here would be to create an extra variable "mapping_command", which will contain "fastp {fastq1} {fastq2} | bwa mem -" if trimming is on, and "bwa mem {fastq1} {fastq2}" if trimming is off. You could then plug this variable into the main command, the same way {keep_unparsed_bams_command} is currently used!